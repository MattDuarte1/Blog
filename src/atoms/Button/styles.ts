import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme, disabled }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 20px 23px 19px;
    gap: 10px;
    border: none;
    cursor: ${disabled ? 'not-allowed' : 'pointer'};

    min-width: 145px;
    height: 67px;
    font-weight: 700;
    color: #ffff;

    background: ${theme.colors.blue};
    border-radius: 8px;
    transition: filter 0.4s ease;

    filter: ${disabled && 'brightness(0.7)'};

    &:hover {
      filter: ${!disabled && 'brightness(1.1)'};
    }
  `}
`;
