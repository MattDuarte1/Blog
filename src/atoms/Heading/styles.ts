import styled, { css, DefaultTheme } from 'styled-components';

interface ContainerProps {
  size?:
    | 'xxsmall'
    | 'xsmall'
    | 'small'
    | 'medium'
    | 'large'
    | 'xlarge'
    | 'xxlarge'
    | 'huge'
    | 'xhuge';
  color:
    | 'white'
    | 'lightGrey1'
    | 'lightGrey2'
    | 'grey'
    | 'darkGrey'
    | 'black'
    | 'blue';
  uppercase: boolean;
}
const MediaFont = {
  large: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      font-size: ${theme.font.sizes.medium};
    }
  `,
  xxlarge: (theme: DefaultTheme) => css`
    @media ${theme.media.lteMedium} {
      font-size: ${theme.font.sizes.xlarge};
    }
    @media ${theme.media.lteSmall} {
      font-size: ${theme.font.sizes.large};
    }
  `,
};

const TitleSize = {
  xxsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxsmall};
  `,
  xsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xsmall};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${MediaFont.large(theme)};
  `,
  xlarge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  xxlarge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxlarge};
    ${MediaFont.xxlarge(theme)};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.huge};
  `,
  xhuge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
  `,
};

const TitleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const TitleColor = (
  color:
    | 'white'
    | 'lightGrey1'
    | 'lightGrey2'
    | 'grey'
    | 'darkGrey'
    | 'black'
    | 'blue',
  theme: DefaultTheme,
) => css`
  color: ${color ? theme.colors[color] : theme.colors.black};
`;

export const Container = styled.h1<ContainerProps>`
  ${({ theme, size, color, uppercase }) => css`
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 700;
    line-height: 1.2;
    text-transform: capitalize;

    ${TitleSize[size](theme)}
    ${TitleCase(uppercase)}
    ${TitleColor(color, theme)}
  `}
`;
