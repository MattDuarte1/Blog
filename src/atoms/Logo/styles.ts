import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 600;
    font-size: ${theme.font.sizes.large};
    line-height: 45px;
    position: relative;
    &::first-letter {
      color: ${theme.colors.blue};
    }

    &::after {
      content: '.blog';
      font-family: ${theme.font.family.default};
      font-style: normal;
      font-weight: 600;
      font-size: ${theme.font.sizes.small};
      color: ${theme.colors.blue};
    }
  `}
`;
