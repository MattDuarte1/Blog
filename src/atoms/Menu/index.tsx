import React, { ComponentProps, FC, forwardRef, HTMLAttributes } from 'react';
import styled, { css } from 'styled-components';
import NextLink from 'next/link';

const StyledLink = styled(NextLink)`
  ${({ theme }) => css`
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 600;
    font-size: ${theme.font.sizes.xsmall};
    line-height: 25px;
    text-transform: capitalize;
    color: ${theme.colors.black};
    cursor: pointer;

    display: flex;
    justify-content: center;
    align-items: center;
    gap: 10px;

    max-width: 100px;
    height: 25px;
  `}
`;

type MenuProps = {
  title: string;
  newTab?: boolean;
  link: string;
};

const Menu = forwardRef<HTMLAnchorElement, MenuProps>(
  ({ title, newTab = false, link, ...props }, ref) => {
    const target = newTab ? '_blank' : '_self';

    return (
      <StyledLink
        ref={ref}
        href={link}
        scroll={false}
        target={target}
        {...props}
      >
        {title}
      </StyledLink>
    );
  },
);

Menu.displayName = 'Menu Component';

export default Menu;
