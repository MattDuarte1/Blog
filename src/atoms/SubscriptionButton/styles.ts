import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.blue};
    border: none;
    border-radius: 25px;
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 0.9rem;
    padding: 1rem;

    cursor: pointer;
  `}
`;
