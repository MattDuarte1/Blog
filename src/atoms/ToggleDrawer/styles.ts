import styled, { css } from 'styled-components';

type IToggle = {
  active?: boolean;
};

export const Toggle = styled.button<IToggle>`
  ${({ theme, active }) => css`
    background-color: #fff;
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.08);
    border-radius: 10px;
    border: none;
    overflow: hidden;
    display: none;
    cursor: pointer;
    justify-content: center;
    align-items: center;
    position: relative;
    height: 70px;
    width: 70px;

    p {
      position: absolute;
      width: 40px;
      height: 4px;
      background-color: ${theme.colors.blue};
      border-radius: 4px;
      transition: 0.5s;
    }

    p:nth-child(1) {
      transform: ${active
        ? 'translateY(0px) rotate(45deg)'
        : 'translateY(-15px)'};
      width: ${active ? '40px' : '25px'};
      left: 15px;
    }

    p:nth-child(2) {
      transform: ${active
        ? 'translateY(0px) rotate(315deg)'
        : 'translateY(15px)'};
      width: ${active ? '40px' : '15px'};
      left: 15px;
    }

    p:nth-child(3) {
      transform: ${active && 'translateX(60px)'};
    }

    @media ${theme.media.lteLarge} {
      display: flex;
    }
  `}
`;
