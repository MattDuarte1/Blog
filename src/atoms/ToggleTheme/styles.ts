import styled, { css } from 'styled-components';

export const Container = styled.div`
  position: fixed;
  bottom: 5rem;
  right: 0rem;
  z-index: 2;
`;

export const Label = styled.label`
  position: relative;
  display: inline-block;
  width: 8.9rem;
  height: 4.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: rgba(0, 0, 0, 0);
`;

export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;
    &:checked + ${Slider} {
      background: ${theme.colors.white};
    }
    &:focus + ${Slider} {
      box-shadow: 0 0 1px ${theme.colors.blue};
    }
    &:checked + ${Slider}:before {
      transform: translateX(4rem);
      background: ${theme.colors.blue};
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${theme.colors.lightGrey2};
    transition: all 300ms ease-in-out;
    border-radius: 2rem;
    box-shadow: 0 0 50px ${theme.colors.lightGrey2};
    &:before {
      content: '';
      position: absolute;
      height: 4.1rem;
      width: 4.1rem;
      left: 0.4rem;
      bottom: 0.2rem;
      background: ${theme.colors.blue};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
