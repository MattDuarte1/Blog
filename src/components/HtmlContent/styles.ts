import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    font-size: calc(${theme.font.sizes.small} + 0.2rem);
    padding: 10rem;
    max-width: 100%;
    line-height: 1.5;

    iframe {
      width: 500px;
    }

    button {
      background-color: ${theme.colors.lightGrey2};
      padding: ${theme.spacings.xsmall};
      cursor: pointer;
      border: none;
    }

    p {
      margin: ${theme.spacings.medium} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.grey};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }
    a:hover {
      filter: brightness(50%);
    }
    code {
      font-family: monospace;
      color: ${theme.colors.black};
      font-size: ${theme.font.sizes.xsmall};
      background: ${theme.colors.darkGrey};
      padding: 0.2rem;
      margin: 0.2rem;
    }
    pre {
      background: ${theme.colors.darkGrey};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.black};
      margin: ${theme.spacings.medium} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
    pre code {
      color: inherit;
      background: inherit;
    }
    img {
      max-width: 100%;
      height: auto;
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.lightGrey1};
    }
    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }
    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 90%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
      background-color: ${theme.colors.darkGrey};
    }
    table td,
    table th {
      padding: ${theme.spacings.xsmall};
      border: 0.1rem solid ${theme.colors.lightGrey1};
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.lightGrey1};
      color: ${theme.colors.darkGrey};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }
    @media ${theme.media.lteMedium} {
      font-size: 2rem;
      padding: 3rem;

      iframe {
        width: 100%;
      }
    }
  `}
`;
