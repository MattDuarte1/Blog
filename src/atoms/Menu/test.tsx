import { screen } from '@testing-library/react';
import Menu from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<Menu>', () => {
  it('Should render a menu link', () => {
    renderTheme(<Menu title="Inicio" link="" />);
    const menu = screen.getByRole('link', { name: 'Inicio' });

    expect(menu).toHaveAttribute('target', '_self');
    expect(getComputedStyle(menu).cursor).toBe('pointer');
  });

  it('Should render a internal link', () => {
    renderTheme(<Menu title="Inicio" link="#" />);
    const menu = screen.getByRole('link', { name: 'Inicio' });

    expect(menu).toHaveAttribute('href', '#');
    expect(menu).toHaveAttribute('target', '_self');
  });

  it('Should render open in a new tab', () => {
    renderTheme(
      <Menu title="Inicio" link="http://www.google.com" newTab={true} />,
    );
    const menu = screen.getByRole('link', { name: 'Inicio' });

    expect(menu).toHaveAttribute('href', 'http://www.google.com');
    expect(menu).toHaveAttribute('target', '_blank');
  });

  it('Should match snapshot of menu', () => {
    renderTheme(<Menu title="procurar" link="#" newTab={false} />);

    expect(screen.getByRole('link', { name: 'procurar' })).toMatchSnapshot();
  });
});
