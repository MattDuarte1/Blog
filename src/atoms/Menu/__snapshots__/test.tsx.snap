// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu> Should match snapshot of menu 1`] = `
.c0 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 600;
  font-size: 1.617rem;
  line-height: 25px;
  text-transform: capitalize;
  color: #2B2C34;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  max-width: 100px;
  height: 25px;
}

<a
  class="c0"
  href="#"
  target="_self"
>
  procurar
</a>
`;
